### PR TITLE
docs: Move nightly builds

### PR DESCRIPTION
**Pull request by popey**
- [X] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [X] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. feat: or fix:)

This was discussed in a Telegraf planning meeting last week. 
This simply moves the large chunk of nightly builds off to developer documentation, to a specific page.

### CODE REVIEW

**Review by srebhan — CHANGES_REQUESTED** [review]
Hey @popey, good idea. However, I guess you forgot to remove the initial links to the nightly build from README.md, didn't you?

**Review by srebhan — APPROVED** [review]
Looks perfect to me. Thanks @popey!

File: README.md [hunk 1]
```diff
@@ -80,46 +80,7 @@ version.
...
-- [telegraf-nightly_linux_s390x.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_linux_s390x.tar.gz)
-- [telegraf-static-nightly_linux_amd64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-static-nightly_linux_amd64.tar.gz)
-
-OSX - .tar.gz
-- [telegraf-nightly_darwin_amd64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_amd64.tar.gz)
-
-Windows - .zip
-- [telegraf-nightly_windows_i386.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_i386.zip)
-- [telegraf-nightly_windows_amd64.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_amd64.zip)
+[Nightly])(/docs/NIGHTLIES.md) builds are available, generated from the master branch.
```
Comment (sspaink) [inline README.md:83]:
```suggestion
[Nightly](/docs/NIGHTLIES.md) builds are available, generated from the master branch.
```

File: docs/README.md [hunk 2]
```diff
@@ -19,3 +21,4 @@
 [profiling]: /docs/PROFILING.md
 [winsvc]: /docs/WINDOWS_SERVICE.md
 [faq]: /docs/FAQ.md
+[nightlies]: /docs/NIGHLIES.md
```
Comment (sspaink) [inline docs/README.md:24]:
```suggestion
[nightlies]: /docs/NIGHTLIES.md
```

**Review by sspaink — CHANGES_REQUESTED** [review]

**Review by powersj — APPROVED** [review]
I merged @sspaink suggestions and I'm +1. @sspaink can you want to re-review and merge if you are happy with this?

**Review by sspaink — APPROVED** [review]

**Pull request merged**